### PR TITLE
fix: number of stories in summary stats is one less than actual stories

### LINKED ISSUE #7092: Sprint Poker: number of stories in summary stats is one less than actual stories
## Issue - Bug

When a Sprint Poker meeting has ended, the number of stories in the summary stats is one less than the actual number of stories that were estimated.

Captura de Pantalla 2022-08-18 a la(s) 3 36 51 p m

Note that the stats at the top says `8` but the list of stories estimated below totals `9`.

- **How serious is the bug?** P4
- **OS:** Unspecified
- **Browser:** Unspecified

### Acceptance Criteria (optional)
- Number of stories in the summary stats is equal to the number of stories estimated

**Estimated effort:** 1 hour to triage. More if root cause is already identified.

### PR DESCRIPTION
**Pull request by 87prashant**
# Description

Fixes/Partially Fixes #7092

## Demo
Screenshot 2022-09-30 234026

## Final checklist

- [x] I checked the [code review guidelines](../docs/codeReview.md)
- [ ] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [x] I have performed a self-review of my code, the same way I'd do it for any other team member
- [x] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [x] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [ ] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [x] PR title is human readable and could be used in changelog

### CODE REVIEW

**Review by nickoferrall — CHANGES_REQUESTED** [review]
Thanks for the PR! The main bug in that issue is the number of stories aren't being calculated correctly because the final story isn't being [marked as complete](https://github.com/ParabolInc/parabol/issues/7092#issuecomment-1247194705). I think we should fix that before closing the linked issue

**Review by nickoferrall — CHANGES_REQUESTED** [review]
It's not working for me - if I vote on three stories, the summary only counts two:

story-count

Is it working for you?

**Comment by 87prashant** [comment]
@nickoferrall 
Yes, but I did `yarn start` also to make it work.

File: packages/server/graphql/mutations/endSprintPoker.ts [hunk 1]
```diff
@@ -64,13 +64,13 @@ export default {
     if (!currentStageRes) {
       return standardError(new Error('Cannot find facilitator stage'), {userId: viewerId})
     }
+    const {stage} = currentStageRes
+    const phase = getMeetingPhase(phases)
+    stage.isComplete = true
```
Comment (87prashant) [inline packages/server/graphql/mutations/endSprintPoker.ts:69]: After these changes, I also did `yarn build`

**Review by nickoferrall — CHANGES_REQUESTED** [review]
This still isn't working for me. @MindEvo would you mind testing this PR to see if it works for you, please?

**Comment by 87prashant** [comment]
It is only working for me, that's weird.
@nickoferrall 
Demo video:

https://user-images.githubusercontent.com/106697681/195330917-d5792c72-4915-4613-ada3-a0648bd5c76a.mp4

**Comment by 87prashant** [comment]
@nickoferrall 
> Is the Story Count always correct for you?

Yes, I ran over 20 meetings and everyone had positive results. It seems like it is counting the 'completed stage' before marking the stage—on which we click the end meeting button—complete.
I don't know if it is related: You used the GitHub issues (in your video) and I used the Parabol issues as stories.

Also, what made you think or how did you know that the fix might have the bug, so, you should test it in a lot of meetings because I only checked it in 2-3 meetings and thought it works. Is it part of the process of how you test things or something that made you suspicious during reviewing? I am just curious and want to improve my process.

**Comment by nickoferrall** [comment]
Thanks, @MindEvo!

@87prashant 
> how did you know that the fix might have the bug, so, you should test it in a lot of meetings

I wanted to test several meetings to make sure that it was fixed - nothing to do with the code, I just noticed after 2/3 meetings that the count was incorrect. 

Have you been able to reproduce the bug that @MindEvo shared https://github.com/ParabolInc/parabol/pull/7241#issuecomment-1276556851?

**Comment by 87prashant** [comment]
@nickoferrall 
> Have you been able to reproduce the bug that @MindEvo shared https://github.com/ParabolInc/parabol/pull/7241#issuecomment-1276556851?

No, even if I don't vote on any of the stages, still I get the correct result.

@MindEvo 
> I think it's counting the stories as completed after loading them, instead of waiting for them to be scored.

Correct me if I am wrong. I think it should count the stories as completed if we visit them regardless of vote/score.

**Comment by 87prashant** [comment]
@Dschoordsch could you also help in reviewing this PR?

File: packages/server/graphql/mutations/endSprintPoker.ts (hunk 1 above)
Comment (Dschoordsch) [reply packages/server/graphql/mutations/endSprintPoker.ts:69]: +1 I ran it through the debugger and this part of the fix works. In my test I had 3 stages, completed 1 and ended on 2. However when I stepped into this function, 1 wasn't completed so I ended up with a miscount here.

-1 Please move stage.endAt here as well to keep the stage modifying statements together.

**Review by Dschoordsch — CHANGES_REQUESTED** [review]

**Comment by 87prashant** [comment]
Ok, thanks!
@Dschoordsch

**Comment by Dschoordsch** [comment]
To be clear, let's focus on fixing this current bug first instead of investigating why sometimes the stage was not completed. We should do that in a separate PR.

**Review by Dschoordsch — APPROVED** [review]
Did a smoke test

**Comment by Dschoordsch** [comment]
Thanks for the PR!

**Pull request merged**